zebra-network/src/peer_set/initialize.rs

```diff
@@ -199,8 +199,16 @@ where
     );
     let _ = candidates.update_initial(active_initial_peer_count).await;
 
-    // TODO: reduce demand by `active_outbound_connections.update_count()` (#2902)
-    for _ in 0..config.peerset_initial_target_size {
+    // Reduce demand by the last `active_outbound_connections.update_count()` output.
+    let demand_count = match config
+        .peerset_initial_target_size
+        .checked_sub(active_outbound_connections.update_count())
+    {
+        Some(count) => count,
+        None => active_initial_peer_count,
```

Comment:
```suggestion
        None => 0,
```

It seems that if a overflow occurs, then we already have `peerset_initial_target_size` or more connections, and should not open more.

(And in that case you can also `saturating_sub` instead and remove the `match`)

Reply:
My mistake, thanks for catching. I made this change in https://github.com/ZcashFoundation/zebra/pull/2969/commits/1571d71f81dd77f78bab267844dd173d32933b38